flags.go

```diff
@@ -0,0 +1,52 @@
+package fangs
+
+import (
+	"reflect"
+
+	"github.com/spf13/pflag"
+)
+
+type FlagAdder interface {
+	AddFlags(flags *pflag.FlagSet)
+}
+
+func AddFlags(flags *pflag.FlagSet, structs ...any) {
+	for _, o := range structs {
+		addFlags(flags, false, o)
+	}
+}
+
+func addFlags(flags *pflag.FlagSet, skip bool, o any) {
+	if !skip {
+		invokeAddFlags(flags, o)
+	}
+
+	v, t := base(reflect.ValueOf(o))
+
+	if isStruct(t) {
+		for i := 0; i < t.NumField(); i++ {
+			f := t.Field(i)
+			v := v.Field(i)
+			v = v.Addr()
+			if !v.CanInterface() {
+				continue
+			}
+
+			addFlags(flags, f.Anonymous, v.Interface())
```

Comment:
actually, i don't think I understand why we'd want to skip this for an anonymous field, can you expound a bit here?

Reply:
(The more I think about it, I think it's important to not skip embedded structs)

Reply:
> skip is specifically used for skipping duplicate calls for embedded structs

the problem being to only call it if there are multiple embeddings that implement the target interface, and to skip it if there is only one since it will be invoked on the parent

Reply:
It took a while to wrangle this, but I think I've got it sorted